src/genomic_features/ensembl/ensembldb.py

```diff
@@ -27,13 +41,66 @@ def annotation(species: str, version: str | int):
     EnsemblDB
         The annotation database.
     """
-    return EnsemblDB(
-        ibis.sqlite.connect(
-            retrieve_annotation(
-                ENSEMBL_URL_TEMPLATE.format(species=species, version=version)
+    try:
+        ensdb = EnsemblDB(
+            ibis.sqlite.connect(
+                retrieve_annotation(
+                    ENSEMBL_URL_TEMPLATE.format(species=species, version=version)
+                )
             )
         )
-    )
+    except HTTPError as err:
+        if err.response.status_code == 404:
+            raise ValueError(
+                f"No Ensembl database found for {species} v{version}. Check available versions with `genomic_features.ensembl.list_versions`."
+            ) from err
+        else:
+            raise HTTPError from err
+    return ensdb
+
+
+def list_versions(species: str) -> DataFrame:
+    """List available Ensembl versions for a species.
+
+    Parameters
+    ----------
+    species
+        The species name. E.g. Hsapiens for human, Mmusculus for mouse.
+
+    Returns
+    -------
+    DataFrame
+        A table of available versions in EnsDb for species of interest.
+    """
+    ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
+
+    # Get latest AnnotationHub timestamp
+    timestamp = requests.get(TIMESTAMP_URL).text
+    ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
+    latest_ts = Timestamp(timestamp).replace(tzinfo=None)
+    cached_ts = ahdb.table("timestamp").execute()["timestamp"][0]
+    if latest_ts != cached_ts:
+        cached_db = glob.glob(
+            os.path.join(pooch.os_cache("genomic-features"), "*annotationhub.sqlite3")
+        )[0]
+        os.remove(cached_db)
+        ahdb = ibis.sqlite.connect(retrieve_annotation(ANNOTATION_HUB_URL))
+
+    version_table = ahdb.table("rdatapaths").filter(_.rdataclass == "EnsDb").execute()
+    version_table = version_table[
+        version_table["rdatapath"].str.contains(f"EnsDb.{species}.v")
+    ]
+    # check that species exists
+    if version_table.shape[0] == 0:
+        raise ValueError(
+            f"No Ensembl database found for {species}. Check species name."
+        )
```

Comment:
It would be nice to get a list of available species as well. We could remove filtering by species and just return the full table?